server/server.go

```diff
@@ -312,16 +323,39 @@ func New(cfg *config.Config) (http.Handler, error) {
 		srv.bkOrder = append(srv.bkOrder, be.Id)
 	}
 
+	var repoNames []string
 	for _, r := range srv.config.IndexConfig.Repositories {
 		srv.repos[r.Name] = r
+		repoNames = append(repoNames, r.Name)
 	}
 
+	// Sort in descending order of length so most specific match is selected by regex engine
```

Comment:
Let's refactor this logic into `func buildRepoRegex(names []string) regexp.Regexp` and write some tests.